src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -54,6 +54,9 @@
  * <p>Enums by definition are static implicit subclasses of java.lang.Enum&#60;E&#62;.
  * So, the <code>static</code> modifier on the enums is redundant. In addition,
  * if enum is inside of interface, <code>public</code> modifier is also redundant.
+ * Enums can contain abstract methods which can be overriden by the declared fields.
+ * Since they can be overridden, final methods in these types are not marked as redundant
+ * even though they can't be extended by other classes/enums.
```

Comment:
I failed to understand this .... till I read examples in issue. We need to think about this description one more time.